plugins/vsphere/pkg/command/source/create.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2020 VMware, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package source
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+
+	"github.com/vmware-tanzu/sources-for-knative/pkg/apis/sources/v1alpha1"
+	"github.com/vmware-tanzu/sources-for-knative/pkg/vsphere"
+	"github.com/vmware-tanzu/sources-for-knative/plugins/vsphere/pkg"
+)
+
+func NewSourceCreateCommand(clients *pkg.Clients, opts *Options) *cobra.Command {
+	result := cobra.Command{
+		Use:   "create",
+		Short: "Create a vSphere source to react to vSphere events",
+		Long:  "Create a vSphere source to react to vSphere events",
+		Example: `# Create the source in the default namespace, sending events to the specified sink URI
+kn vsphere source create --name vc-01-source --address https://my-vsphere-endpoint.local --skip-tls-verify --secret-ref vsphere-credentials --sink-uri http://where.to.send.stuff
+
+# Create the source in the specified namespace, sending events to the specified service
+kn vsphere source create --namespace ns --name vc-01-source --address https://my-vsphere-endpoint.local --skip-tls-verify --secret-ref vsphere-credentials --sink-api-version v1 --sink-kind Service --sink-name the-service-name
+
+# Create the source in the specified namespace, sending events to the specified service with custom checkpoint behavior
+kn vsphere source create --namespace ns --name vc-01-source --address https://my-vsphere-endpoint.local --skip-tls-verify --secret-ref vsphere-credentials --sink-api-version v1 --sink-kind Service --sink-name the-service-name --checkpoint-age 1h --checkpoint-period 30s
+`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if opts.Name == "" {
+				return fmt.Errorf("'name' requires a nonempty name provided with the --name option")
+			}
+			if opts.Address == "" {
+				return fmt.Errorf("'address' requires a nonempty address provided with the --address option")
+			}
+			if opts.SecretRef == "" {
+				return fmt.Errorf("'secret-ref' requires a nonempty secret reference provided with the --secret-ref option")
+			}
+			sinkCoordinatesAllEmpty := opts.SinkAPIVersion == "" && opts.SinkKind == "" && opts.SinkName == ""
+			sinkCoordinatesAllSet := opts.SinkAPIVersion != "" && opts.SinkKind != "" && opts.SinkName != ""
+			if opts.SinkURI == "" && sinkCoordinatesAllEmpty ||
+				(!sinkCoordinatesAllEmpty && !sinkCoordinatesAllSet) {
+				return fmt.Errorf("sink requires an URI" +
+					"\nand/or a nonempty API version --sink-api-version option," +
+					"\nwith a nonempty kind --sink-kind option," +
+					"\nand with a nonempty name with the --sink-name")
+			}
+
+			// verify supported datacontentencoding schemes
+			validEncodings := map[string]struct{}{
+				"xml":  {},
+				"json": {},
+			}
```

Comment:
Consider using the [k8s `sets.String`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#String) here.